front/pages/api/w/[wId]/data_sources/[name]/databases/csv.ts

```diff
@@ -212,12 +212,44 @@ async function rowsFromCsv(
   csv: string
 ): Promise<Result<CoreAPIDatabaseRow[], APIError>> {
   // Detect the delimiter.
-  const delimiters = [",", ";", "\t"];
+  const firstTwoLines = csv.split("\n").slice(0, 2);
+  if (firstTwoLines.length < 2) {
+    return new Err({
+      type: "invalid_request_error",
+      message: `CSV is empty.`,
```

Comment:
CSV must contain at least 2 records?

Reply:
There's the header.

Also this PR is no good, because it doesn't work with escaped separators and cells that contain the separator char but are wrapped in `"`. I'm working on something better that uses the parsing library to parse the first 2 lines.

Reply:
I think the counting and checking for equality mitigates the escaping issue to some large extent?

Reply:
I am probably missing something but I don't see how ?

```
hello,world
"I, henry, am part of the first column",I am the second column
```

Reply:
yeah it's def not perfect.

Then you could try parsing 2-3 lines with each parameter and use the one that has equal number of elements.

Reply:
Yes that's the plan (first 2 lines, so header + 1 row)